LiteCore/Storage/SQLiteDataFile.cc

```diff
@@ -499,11 +541,45 @@ path.path().c_str());
         }
     }
 
+    void SQLiteDataFile::collectBlobs(string storeName, unordered_set<string> &digests) {
+        stringstream s;
+        s << "SELECT body FROM kv_" << storeName << " WHERE (flags & " <<
+            (int)DocumentFlags::kHasAttachments << ") != 0 AND (flags & " <<
+            (int)DocumentFlags::kDeleted << ") == 0";
+        
+        string sql = s.str();
+        compile(_collectBlobStmt, sql.c_str());
+        UsingStatement u(_collectBlobStmt);
+        while(_collectBlobStmt->executeStep()) {
+            slice sl = SQLiteKeyStore::columnAsSlice(_collectBlobStmt->getColumn(0));
+            slice body = RawRevision::getCurrentRevBody(sl);
+            const Dict* root = Value::fromTrustedData(body)->asDict();
+            findBlobs(root, digests, documentKeys());
+        }
+    }
+    
+    void SQLiteDataFile::removeUnusedBlobs(const unordered_set<string> &used) {
+        FilePath attsPath(filePath().dirName() + "Attachments", "");
```

Comment:
There's a BlobStore class that manages access to blobs. You should go through that instead of groping into the directory here.